### PR TITLE
config: add NewFromKiteKey

### PR DESCRIPTION
**Pull request by rjeczalik**

### CODE REVIEW

**Comment by sent-hil** [comment]
@rjeczalik what's the reason for this?

**Comment by rjeczalik** [comment]
@sent-hil To read `/etc/kite.key` without `os.Putenv("KITE_HOME", "/etc/kite.key");  kitekey.Parse()`.

File: config/config.go [hunk 1]
```diff
@@ -47,6 +48,21 @@ func New() *Config {
...
 }
 
+// NewFromKiteKey parses the given kite key file and gives a new Config value.
+func NewFromKiteKey(file string) (*Config, error) {
+	key, err := kitekey.ParseFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	var c Config
```
Comment (sent-hil) [inline config/config.go]: any reason this is not `*Config`...don't usually see non pointer declaration and return with `&`

**Comment by sent-hil** [comment]
LGTM, can merge when tests pass

File: config/config.go (hunk 1 above)
Comment (rjeczalik) [inline config/config.go]: less keystrokes ;-)

Comment (sent-hil) [inline config/config.go]: only if * and & somehow take more keystrokes in your keyboard :)

Comment (rjeczalik) [inline config/config.go]:
```
var c Config&
c := &Config{}
c := new(Config)
```

It's 1 to 3 keystrokes shorter :sunglasses: :sunglasses: :sunglasses: :trollface:

Comment (sent-hil) [inline config/config.go]: image

**Comment by rjeczalik** [comment]
The failure on master is caused by etcd using API from go1.5 - `os.LookupEnv`, needs to be pinned to 1.4-compatible release,

**Comment by sent-hil** [comment]
@rjeczalik could you do that when you get the chance please?

**Pull request merged**